pkgs/tools/package-management/home-manager/default.nix

```diff
@@ -0,0 +1,45 @@
+#Taken directly from
+#https://github.com/rycee/home-manager/blob/9c1b3735b402346533449efc741f191d6ef578dd/home-manager/default.nix
+
+{ pkgs,bash,coreutils,less,stdenv,makeWrapper,fetchFromGitHub
+
+  # Extra path to the Home Manager modules. If set then this path will
+  # be tried before `$HOME/.config/nixpkgs/home-manager/modules` and
+  # `$HOME/.nixpkgs/home-manager/modules`.
+, modulesPath ? null
+}:
+
+stdenv.mkDerivation rec{
+  name = "home-manager-${version}";
+  version = "2017-10-11";
+  src=fetchFromGitHub{
+    owner = "rycee";
+    repo = "home-manager";
+    rev = "7e6f3364bcf0a0ec838aa4853f550a9a7b5ed027";
+    sha256 = "04wl0jpha19fsnyc5a8y4pkss9by468k0i5w5bjswnaz792yji6v";
+  };
+  buildInputs=[makeWrapper];
+  phases = [ "installPhase" ];
```

Comment:
You probably in general don't want to override ```phases```. Instead specifically exclude the phases you don't need. In this case that will probably be ```dontBuild = true;```.